cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/security/jwt/JwtLoginService.java

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright 2020 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.cruisecontrol.servlet.security.jwt;
+
+import com.linkedin.kafka.cruisecontrol.servlet.security.UserStoreAuthorizationService;
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.JWSObject;
+import com.nimbusds.jose.JWSVerifier;
+import com.nimbusds.jose.crypto.RSASSAVerifier;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.SignedJWT;
+import org.eclipse.jetty.security.DefaultIdentityService;
+import org.eclipse.jetty.security.IdentityService;
+import org.eclipse.jetty.security.LoginService;
+import org.eclipse.jetty.security.authentication.AuthorizationService;
+import org.eclipse.jetty.server.UserIdentity;
+import org.eclipse.jetty.util.component.AbstractLifeCycle;
+import org.eclipse.jetty.util.component.LifeCycle;
+
+import javax.security.auth.Subject;
+import javax.servlet.ServletRequest;
+import javax.servlet.http.HttpServletRequest;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.security.interfaces.RSAPublicKey;
+import java.text.ParseException;
+import java.time.Clock;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static com.linkedin.kafka.cruisecontrol.servlet.security.jwt.JwtAuthenticator.JWT_LOGGER;
+
+/**
+ * <p>This class validates a JWT token. The token must be cryptographically encrypted an It uses an RSA public key for
+ * validation that is expected to be stored in a PEM formatted file.</p>
+ * <p>This class implements {@link AbstractLifeCycle} which means it is a managed bean, its lifecycle will be managed
+ * by Jetty. It's {@link AuthorizationService} can also be an {@link AbstractLifeCycle} in which case it delegates to
+ * this class, so opening and closing connections should be done by implementing the {@link AbstractLifeCycle} interface's
+ * {@link #doStart()} and {@link #doStop()} methods respectively. For a simple example see
+ * {@link UserStoreAuthorizationService}.</p>
+ * <p>The login service also validates expiration time of the token and it expects the token to contain the expiration
+ * in Unix epoch time format in UTC.</p>
+ */
+public class JwtLoginService extends AbstractLifeCycle implements LoginService {
+
+  private final AuthorizationService _authorizationService;
+  private IdentityService _identityService = new DefaultIdentityService();
```

Comment:
Nit: For code style consistency, can we move this initialization to the constructor?

Reply:
Sure